### PR TITLE
Added new page under Data types and IO section for tensorflow_types in flyte documentation

### LINKED ISSUE #4065: [Docs] tensorflow types: create flytesnacks example page
### Description

Add a page to the [Data Types and IO](https://docs.flyte.org/projects/cookbook/en/latest/auto_examples/data_types_and_io/index.html) section in the user guide that documents all [available TensorFlow types](https://github.com/flyteorg/flytekit/tree/master/flytekit/extras/tensorflow) and explains how they work.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by sumana-2705**
## Tracking issue
Closes #4065

## Why are the changes needed?

- **User Understanding**: Enhances clarity on how to use TensorFlow types within Flyte.
- **Integration Guidance**: Provides step-by-step instructions for integrating TensorFlow models and datasets.
- **Error Reduction**: Helps users avoid common pitfalls and integration errors.
- **Improved Documentation**: Fills a gap in the existing documentation for better usability.
- **Increased Efficiency**: Streamlines the workflow for users working with TensorFlow in Flyte.

## What changes were proposed in this pull request?

This PR introduces a new page in the Data Types and IO section of the user guide, documenting all available TensorFlow types in Flyte. It includes detailed explanations of the `TensorFlowModel`, `TFRecordFile`, and `TFRecordsDirectory`, along with the `TFRecordDatasetConfig` configuration class. 

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All commits are signed-off.

## Docs link

[Data Types and IO](https://docs.flyte.org/en/latest/user_guide/data_types_and_io/index.html)

### CODE REVIEW

**Comment by davidmirror-ops** [comment]
@neverett do you think the failing test here has to do with content relocation?

**Comment by neverett** [comment]
@davidmirror-ops I don't think so, since the "Data types and I/O" section wasn't moved as part of the Flyte docs overhaul. It looks like the readthedocs test is passing and a bunch of other tests are pending -- can you point me to the test failure you're seeing?

**Review by davidmirror-ops — APPROVED** [review]
Thank you!

**Comment by sumana-2705** [comment]
@davidmirror-ops  I'm so sorry, I have just tried to fix the errors in the tests. I don't know that it would cancel the approval. I'm unaware of that. Please don't mind my mistake.

File: docs/user_guide/data_types_and_io/tensorflow_type.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+This document outlines the TensorFlow types available in Flyte, which facilitate the integration of TensorFlow models and datasets in Flyte workflows.
+
+### Import necessary libraries and modules
+```{rli} https://raw.githubusercontent.com/flyteorg/flytesnacks/refs/heads/master/examples/data_types_and_io/data_types_and_io/tensorflow_type.py
+:caption: data_types_and_io/tensorflow_type.py
+:lines: 2-14
+```
+
+## Tensorflow Model
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/tensorflow_type.md:17]:
```suggestion
## Tensorflow model
```

File: docs/user_guide/data_types_and_io/tensorflow_type.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+```{note}
+To clone and run the example code on this page, see the [Flytesnacks repo][flytesnacks].
+```
+```{rli} https://raw.githubusercontent.com/flyteorg/flytesnacks/refs/heads/master/examples/data_types_and_io/data_types_and_io/tensorflow_type.py
+:caption: data_types_and_io/tensorflow_type.py
+:lines: 16-33
+```
+
+## TFRecord Files
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/tensorflow_type.md:38]:
```suggestion
## TFRecord files
```

File: docs/user_guide/data_types_and_io/tensorflow_type.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+### Usage
+The `TensorFlowRecordFileTransformer` enables you to work with single TFRecord files, making it easy to read and write data in TensorFlow's TFRecord format.
+
+```{rli} https://raw.githubusercontent.com/flyteorg/flytesnacks/refs/heads/master/examples/data_types_and_io/data_types_and_io/tensorflow_type.py
+:caption: data_types_and_io/tensorflow_type.py
+:lines: 35-45
+```
+
+## TFRecord Directories
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/tensorflow_type.md:55]:
```suggestion
## TFRecord directories
```

File: docs/user_guide/data_types_and_io/tensorflow_type.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+### Usage
+The `TensorFlowRecordsDirTransformer` allows you to work with directories of TFRecord files, which is useful for handling large datasets that are split across multiple files.
+
+#### Example
+```{rli} https://raw.githubusercontent.com/flyteorg/flytesnacks/refs/heads/master/examples/data_types_and_io/data_types_and_io/tensorflow_type.py
+:caption: data_types_and_io/tensorflow_type.py
+:lines: 47-56
+```
+
+## Configuration Class: `TFRecordDatasetConfig`
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/tensorflow_type.md:74]:
```suggestion
## Configuration class: `TFRecordDatasetConfig`
```

File: docs/user_guide/data_types_and_io/tensorflow_type.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+The `TFRecordDatasetConfig` class is a data structure used to configure the parameters for creating a `tf.data.TFRecordDataset`, which allows for efficient reading of TFRecord files. This class uses the `DataClassJsonMixin` for easy JSON serialization.
+
+### Attributes:
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/tensorflow_type.md:77]:
```suggestion
### Attributes
```

**Review by neverett — APPROVED** [review]
Added a few suggestions, but otherwise this looks good to me. Feel free to request another review from me after you're done making changes!

**Comment by sumana-2705** [comment]
@neverett I have included the changes can you please review it now.

**Review by neverett — APPROVED** [review]

**Comment by sumana-2705** [comment]
@davidmirror-ops, I have updated my forked repository to the latest commits; could you please proceed with merging this PR?

**Review by davidmirror-ops — APPROVED** [review]

**Pull request merged**